sycl/test/acap/test_aie_device_allocator.cpp

```diff
@@ -0,0 +1,61 @@
+// REQUIRES: acap
+
+// RUN: %acap_clang %s -o %s.bin
+// RUN: %add_acap_result %s.bin
+// RUN: rm %s.bin
+/// Example of using neighbor memory tiles
+
+#include <sycl/sycl.hpp>
+
+#include "triSYCL/vendor/Xilinx/acap/aie/device_allocator.hpp"
+
+#include <cstring>
+#include <iostream>
+#include <vector>
+
+static constexpr unsigned count = 6;
+
+using namespace sycl::vendor::xilinx;
+/// A memory tile has to inherit from acap::aie::memory<AIE, X, Y>
+template <typename AIE, int X, int Y>
+struct tile_memory : acap::aie::memory<AIE, X, Y> {
+  uint32_t arr[count];
+};
+
+template <typename AIE, int X, int Y> struct prog : acap::aie::tile<AIE, X, Y> {
+  using t = acap::aie::tile<AIE, X, Y>;
+  /// The lock function of every tiles is run before the prerun of any other
+  /// tile.
+  bool prerun() {
+    return 1;
+  }
+  void run() {
+#ifdef __SYCL_DEVICE_ONLY__
+    __builtin_memset(&t::mem().arr, 0xff, sizeof(t::mem().arr));
```

Comment:
Could be a range algorithm instead?

Reply:
OK, but why some obscure built-in instead of `std::memset`?

Reply:
So, does it compile with `std::memset`? You already includes `<cstring>`, good news. :-)